aspnetcore/mvc/views/view-compilation.md

```diff
@@ -4,14 +4,91 @@ author: rick-anderson
+
+# [Visual Studio](#tab/visual-studio)
+
+In the **Create a new ASP.NET Core web application** dialog:
+
+1. Select either the **Web Application** or the **Web Application (Model-View-Controller)** project template.
+1. Select the **Enable Razor runtime compilation** check box.
+
+# [.NET Core CLI](#tab/netcore-cli)
+
+Use the `-rrc` or `--razor-runtime-compilation` template option. For example, the following command creates a new Razor Pages project with runtime compilation enabled:
+
+```dotnetcli
+dotnet new webapp --razor-runtime-compilation
+```
+
+---
+
+## Enable runtime compilation to existing project
+
+To enable runtime compilation for all environments in an existing project:
+
+1. Install the [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation/) NuGet package.
+1. Update the project's `Startup.ConfigureServices` method to include a call to <xref:Microsoft.Extensions.DependencyInjection.RazorRuntimeCompilationMvcBuilderExtensions.AddRazorRuntimeCompilation*>. For example:
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRazorPages()
+            .AddRazorRuntimeCompilation();
+
+        // code omitted for brevity
+    }
+    ```
+
+## Conditionally enable runtime compilation in existing project
+
+Runtime compilation can be enabled such that it's only available for local development. Conditionally enabling in this manner ensures that the published output:
+
+* Uses compiled views.
+* Is smaller in size.
+* Doesn't enable file watchers in production.
+
+To enable runtime compilation only in the Development environment:
+
+1. Install the [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation/) NuGet package.
+1. Modify the launch profile `environmentVariables` section in *launchSettings.json*:
+    * Set the `ASPNETCORE_ENVIRONMENT` environment variable to `Development`.
+    * Set the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable to the runtime compilation assembly name.
+
+In the following example, runtime compilation is enabled in the Development environment for the `IIS Express` and `RazorPagesApp` launch profiles:
+
+[!code-json[](~/mvc/views/view-compilation/samples/3.1/launchSettings.json?highlight=15-16,24-25)]
+
+No code is needed in the project's `Startup` class. At runtime, ASP.NET Core searches for an [assembly-level HostingStartup attribute](xref:fundamentals/configuration/platform-specific-configuration#hostingstartup-attribute) in `Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation`&mdash;the assembly specified by the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable. The `HostingStartup` attribute specifies the app startup code to execute. That startup code enables runtime compilation.
```

Comment:
per style guide, avoid mdash
plus the sentence is too long to MT
It's easy to split into two sentences, but I'm not sure you need specified by the `ASPNETCORE_HOSTINGSTARTUPASSEMBLIES` environment variable